ipam/allocator.go

```diff
@@ -696,8 +700,12 @@ func (alloc *Allocator) actorLoop(actionChan <-chan func(), stopChan <-chan stru
 			// Retry things in case messages got lost between here and recipients
 			if alloc.awaitingConsensus {
 				alloc.propose()
-			} else {
-				alloc.tryPendingOps()
+			} else if alloc.havePendingOps() {
+				if alloc.ring.Empty() {
+					alloc.establishRing()
+				} else {
+					alloc.tryPendingOps()
+				}
```

Comment:
`establishRing()` may not actually establish a ring; it may just send out some messages.  And not calling `tryPendingOps()` until we have a ring is the whole point of this change.  We could probably assume that combination is ok, because of what we know about the implementation of the consensus protocol, but I'd rather not.

`if alloc.havePendingOps()` is to defer establishing a ring until there is some need for it; this is primarily useful in the case that we don't know the right number of peers, and are still forming connections.